client/client.go

```diff
@@ -26,21 +26,36 @@ import (
 	"github.com/awnumar/memguard"
 	"github.com/lestrrat-go/jwx/v2/jwa"
 	"github.com/lestrrat-go/jwx/v2/jwk"
-	"github.com/lestrrat-go/jwx/v2/jws"
 
 	"github.com/openpubkey/openpubkey/gq"
 	"github.com/openpubkey/openpubkey/pktoken"
 	"github.com/openpubkey/openpubkey/pktoken/clientinstance"
 	"github.com/openpubkey/openpubkey/util"
+	"github.com/openpubkey/openpubkey/verifier"
 )
 
-// OpkClient is the OpenPubkey client
+// Interface for interacting with the OP (OpenID Provider)
+type OpenIdProvider interface {
+	RequestTokens(ctx context.Context, cicHash string) (*memguard.LockedBuffer, error)
+	Verifier() verifier.ProviderVerifier
```

Comment:
Is there a reason not to use composition for the Verifier here? That way rather than `op.Verifier().Issuer()` you could just call `op.Issuer()`?

Reply:
`Issuer()` is required by the verifier not the client. That's why `op.Verifier().Issuer()` is not used in the code base, just tests for test reasons and not for real-world reasons.

Reply:
The issuer value is required by the client. It needs to the issuer value to look up 


```golang
var (
	key            = []byte("NotASecureKey123")
	issuedAtOffset = 1 * time.Minute
)
const googleIssuer = "https://accounts.google.com"
type GoogleOp struct {
	ClientID        string
```
https://github.com/openpubkey/openpubkey/pull/88/files#diff-d18f44e884f4688a7c606a2716462f72a73686038059b03101a7e72cbcb04307L44

The issuance code in the Op needs to the issuer URI to do the well-known look up and get the authorization_endpoint to initiate authorization.

Reply:
I'm not sure what you're pointing to here, but there is no location in the client that requires this information. I think you might be mistakenly referring to the Google Provider which calls Issuer() unnecessarily since issuer is essentially a class variable and there is no value in calling the function instead.